generator/spring-boot/src/main/java/io/fabric8/maven/generator/springboot/SpringBootGenerator.java

```diff
@@ -52,6 +60,40 @@ public boolean isApplicable(List<ImageConfiguration> configs) {
     }
 
     @Override
+    public List<ImageConfiguration> customize(List<ImageConfiguration> configs) throws MojoExecutionException {
+        // if enabled lets generate a spring devtools token
+        String flag = getProjectProperty("fabric8.spring-devtools.generateToken", null);
+        boolean generateToken = !Strings.isNullOrEmpty(flag) && Configs.asBoolean(flag);
+        if (generateToken || getContext().runningWithGoal("fabric8:watch-spring-boot", "fabric8:watch")) {
```

Comment:
as said above, wouldn't do the lookup here but from the outside, injecting the build mode so to use a `getContext().getBuildMode()` here.